mne/epochs.py

```diff
@@ -2596,29 +2551,43 @@ def get_epoch_annotations(self):
         if self.annotations is None:
             return epoch_annot_list
 
-        for annot in self.annotations:
-            onset_ = annot['onset']
-            duration_ = annot['duration']
-            description_ = annot['description']
-
-            # convert onset to samples and account for first time
-            onset_samp = np.rint(onset_ * self._raw_sfreq +
-                                 self._first_samp).astype(int)
-            duration_samp = np.rint(duration_ * self._raw_sfreq).astype(int)
-
-            # loop through events to see which Epochs this annotation
-            # belongs to based on the onset and duration
-            epoch_index = _get_epoch_index_of_annot(
-                self, onset_samp, duration_samp)
-
-            for eidx in epoch_index:
-                # now convert onset sample relative to the Epoch and convert
-                # back to seconds relative to the Epoch
-                onset_samp_ = onset_samp - events[eidx, 0]
-                onset_ = np.divide(onset_samp_, self._raw_sfreq)
-
-                epoch_annot_list[eidx].append(
-                    (onset_, duration_, description_))
+        # when each epoch and annotation starts/stops
+        # no need to account for first_samp here...
+        epoch_tzeros = events[:, 0] / self._raw_sfreq
+        epoch_starts, epoch_stops = np.atleast_2d(
+            epoch_tzeros) + np.atleast_2d(self.times[[0, -1]]).T
+        # ... because first_samp isn't accounted for here either
+        annot_starts = self._annotations.onset
+        annot_stops = annot_starts + self._annotations.duration
+
+        # get all annotations that have a start point within epoch
+        annot_straddles_epoch_start = np.logical_and(
+            np.atleast_2d(epoch_starts) >= np.atleast_2d(annot_starts).T,
+            np.atleast_2d(epoch_starts) < np.atleast_2d(annot_stops).T)
+
+        # get all annotations that have an endpoint within epoch
+        annot_straddles_epoch_end = np.logical_and(
+            np.atleast_2d(epoch_stops) > np.atleast_2d(annot_starts).T,
+            np.atleast_2d(epoch_stops) <= np.atleast_2d(annot_stops).T)
+
+        # get all annotations that are fully within an epoch
+        annot_fully_within_epoch = np.logical_and(
+            np.atleast_2d(epoch_starts) <= np.atleast_2d(annot_starts).T,
+            np.atleast_2d(epoch_stops) >= np.atleast_2d(annot_stops).T)
+        all_cases = (annot_straddles_epoch_start +
+                     annot_straddles_epoch_end +
+                     annot_fully_within_epoch)
+
+        for annot_ix, epo_ix in zip(*np.nonzero(all_cases)):
+            this_annot = self._annotations[annot_ix]
+            this_tzero = epoch_tzeros[epo_ix]
+
+            # get all annotations for this Epoch and offset the onset
+            # relative to the start of the Epoch
```

Comment:
this comment is inaccurate (specifically the "get all annotations" part; it's within a loop so it's only "getting" one annotation at a time.